tests/testMeasurementFramework.py

```diff
@@ -0,0 +1,100 @@
+# the GNU General Public License along with this program. If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import unittest
+
+import lsst.utils.tests
+import lsst.meas.base.tests
+from lsst.meas.base.tests import AlgorithmTestCase
+from lsst.meas.extensions.simpleShape import SimpleShapeResultKey
+
+class SimpleShpaeMFTestCase(AlgorithmTestCase):
+    def setUp(self):
+        self.bbox = lsst.afw.geom.Box2I(lsst.afw.geom.Point2I(-20, -30),
+                                        lsst.afw.geom.Extent2I(240, 160))
+        self.dataset = lsst.meas.base.tests.TestDataset(self.bbox)
+        self.dataset.addSource(100000.0, lsst.afw.geom.Point2D(149.9, 50.3),
+                               lsst.afw.geom.ellipses.Quadrupole(8, 9, 3))
+
+        self.expectedKeys = ['ext_simpleShape_SimpleShape_IxSigma',
+                             'ext_simpleShape_SimpleShape_Ix_Iy_Cov',
+                             'ext_simpleShape_SimpleShape_IxxSigma',
+                             'ext_simpleShape_SimpleShape_Ixx_Ix_Cov',
+                             'ext_simpleShape_SimpleShape_Ixx_Ixy_Cov',
+                             'ext_simpleShape_SimpleShape_Ixx_Iy_Cov',
+                             'ext_simpleShape_SimpleShape_Ixx_Iyy_Cov',
+                             'ext_simpleShape_SimpleShape_IxySigma',
+                             'ext_simpleShape_SimpleShape_Ixy_Ix_Cov',
+                             'ext_simpleShape_SimpleShape_Ixy_Iy_Cov',
+                             'ext_simpleShape_SimpleShape_IySigma',
+                             'ext_simpleShape_SimpleShape_IyySigma',
+                             'ext_simpleShape_SimpleShape_Iyy_Ix_Cov',
+                             'ext_simpleShape_SimpleShape_Iyy_Ixy_Cov',
+                             'ext_simpleShape_SimpleShape_Iyy_Iy_Cov',
+                             'ext_simpleShape_SimpleShape_flag',
+                             'ext_simpleShape_SimpleShape_x',
+                             'ext_simpleShape_SimpleShape_xx',
+                             'ext_simpleShape_SimpleShape_xy',
+                             'ext_simpleShape_SimpleShape_y',
+                             'ext_simpleShape_SimpleShape_yy']
+
+        def tearDown(self):
+            del self.bbox
+            del self.dataset
+
+        def testKeys(self):
+            """
+            Test that Simple shape will run as a plugin, a functor key can be
+            created, and result object can be created. Testing of the algorithm
+            is left to testSimpleShape.py
+            """
+
+            task = self.makeSingleFrameMeasurementTask("ext_simpleShape_SimpleShape")
+            exposure, catalog = self.dataset.realize(10.0, task.schema)
+            # Check the keys are in the schema
+            schemaKeys = catalog.schema.getNames()
+            self.assertTrue(sum([x in schemaKeys for x in self.expectedKeys]) == len(self.expectedKeys))
```

Comment:
It would be nice to see the missing keys if the test fails. Also sum over booleans is not very clear and assertTrue gives lousy diagnostics when the test fails and should be avoided unless you really are checking some naturally boolean result. Here's a suggested alternative:

    self.expectedKeySet = set([...]) # to replace self.expectedKeys = [...] above
    ...
    foundKeySet = set(catalog.schema.getNames()) & expectedKeySet
    missingKeySet = expectedKeySet - foundKeySet
    self.assertEqual(missingKeySet, set())